var/spack/repos/builtin/packages/babelflow/package.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Babelflow(CMakePackage):
+    """BabelFlow is an Embedded Domain Specific Language to describe
+       algorithms using a task graph abstraction which allows them to be
+       executed on top of one of several available runtime systems."""
+
+    homepage = "https://github.com/sci-visus/BabelFlow"
+    url      = "https://github.com/sci-visus/BabelFlow/archive/ascent.zip"
+
+    maintainers = ['spetruzza']
+
+    version('develop',
+            git='https://github.com/sci-visus/BabelFlow.git',
+            branch='ascent',
+            commit='62e0eae8b2ff28094ec03f0c2496e579dda794ab',
+            submodules=True,
+            preferred=True)
+
+    depends_on('mpi')
+
+    variant("shared", default=True, description="Build Babelflow as shared libs")
+
+    def cmake_install(self, spec, prefix):
+        if "+shared" in spec:
+            cmake_args.append('-DBUILD_SHARED_LIBS=ON')
+        else:
+            cmake_args.append('-DBUILD_SHARED_LIBS=OFF')
+
+        make()
+        make('install')
```

Comment:
```suggestion
    def cmake_args(self):
        args = [
            '-DBUILD_SHARED_LIBS:BOOL={0}'.format(
                'ON' if '+shared' in spec else 'OFF')]
                
        return args
```

Reply:
Sounds good! 
I think I was using that property as I have seen in some other package example, it's strange that I did not incur in an errors when building... 
Anyway your solution seems great.

Reply:
> I think I was using that property as I have seen in some other package example

There are 0 packages in Spack that use this property. If you find any docs that use that, let me know and we can fix them.

> it's strange that I did not incur in an errors when building...

That's because the function can be defined without an issue, but won't be used. Spack will just use the default cmake/make/make install steps.